layers.go

```diff
@@ -117,6 +117,15 @@ type Layer struct {
 
 	// ReadOnly is true if this layer resides in a read-only layer store.
 	ReadOnly bool `json:"-"`
+
+	// BigDataNames is a list of names of data items that we keep for the
+	// convenience of the caller.  They can be large, and are only in
+	// memory when being read from or written to disk.
+	BigDataNames []string `json:"big-data-names,omitempty"`
+
+	// BigDataSizes maps the names in BigDataNames to the sizes of the data
+	// that has been stored, if they're known.
+	BigDataSizes map[string]int64 `json:"big-data-sizes,omitempty"`
```

Comment:
Just reading the API documentation, I would expect `BigDataSizes` to be included in `ImageSize`. (Otherwise it’s not even clear to me there needs to be a separate faster-to-access record of the sizes.)

Reply:
I've only added it because it was present in the image API.  I am fine with dropping it

Reply:
Let's drop it.

Reply:
Why _shouldn’t_ the size be included in `ImageSize`?

Reply:
> Let's drop it.

dropped and pushed a new version.

Reply:
> Why _shouldn’t_ the size be included in `ImageSize`?

it is not used at the moment, we can add it if there is any user of it

Reply:
The Podman repo suggests that `store.ImageSize` is called for “disk usage statistics” (`df`?), “inspect”, “prune” and “tree”; I didn’t trace the full call stack, it may be exposed via other features or it might be dead code.

OTOH it’s true that c/image’s `types.Image.Size()` implementation does its own calculation which does not currently benefit from extending `store.ImageSize`.